gsy_framework/sim_results/electric_blue/aggregate_results.py

```diff
@@ -0,0 +1,145 @@
+from collections import defaultdict
+from dataclasses import dataclass, field, fields
+from typing import Dict, List
+
+from pendulum import DateTime
+
+from gsy_framework.enums import AggregationResolution, AvailableMarketTypes
+
+# Used by forward markets; the following dictionary defines
+# what aggregations are needed for each market type.
+MARKET_RESOLUTIONS = {
+    AvailableMarketTypes.YEAR_FORWARD: [AggregationResolution.RES_1_MONTH],
+    AvailableMarketTypes.MONTH_FORWARD: [AggregationResolution.RES_1_DAY],
+    AvailableMarketTypes.WEEK_FORWARD: [AggregationResolution.RES_1_DAY]
+}
+
+
+@dataclass
+class ForwardDeviceStats:  # pylint: disable=too-many-instance-attributes
+    """Hold forward device statistics for each timeslot. This class is used to first
+    calculate the current results of device in the market and then used to merge it
+    with the global results."""
+
+    timeslot: str
+    device_uuid: str
+    current_timeslot: DateTime
+
+    # SELLER
+    total_energy_produced: float = 0
+    total_sell_trade_count: int = 0
+    total_energy_sold: float = 0
+    total_earned_eur: float = 0
+
+    # BUYER
+    total_energy_consumed: float = 0
+    total_buy_trade_count: int = 0
+    total_energy_bought: float = 0
+    total_spent_eur: float = 0
+
+    open_offers: List[Dict] = field(default_factory=list)
```

Comment:
AFAIU, you do not need this, as there is no default list, right?
Wouldn't the following be sufficient?
```suggestion
    open_offers: List[Dict] = []
```

Reply:
Well, the @dataclss decorator doesn't allow you to use `open_offers: List[Dict] = []`. It needs a factory object.

Reply:
Ahh, I seen, thanks! Dataclasses are still new to me.